planner/core/logical_plan_builder.go

```diff
@@ -725,37 +724,37 @@ func (p *LogicalJoin) setPreferredJoinType4PhysicalOp() {
 				preferJoinType |= preferLeftAsINLHJInner
 			}
 			if rightHintInfo != 0 {
-				preferJoinType |= preferLeftAsINLHJInner
+				preferJoinType |= preferRightAsINLHJInner
 			}
 		case preferINLMJ:
 			if leftHintInfo != 0 {
 				preferJoinType |= preferLeftAsINLMJInner
 			}
 			if rightHintInfo != 0 {
-				preferJoinType |= preferLeftAsINLMJInner
+				preferJoinType |= preferRightAsINLMJInner
 			}
 		case preferHJBuild:
 			if leftHintInfo != 0 {
 				preferJoinType |= preferLeftAsHJBuild
 			}
 			if rightHintInfo != 0 {
-				preferJoinType |= preferLeftAsHJBuild
+				preferJoinType |= preferRightAsHJBuild
 			}
 		case preferHJProbe:
 			if leftHintInfo != 0 {
 				preferJoinType |= preferLeftAsHJProbe
 			}
 			if rightHintInfo != 0 {
-				preferJoinType |= preferLeftAsHJProbe
+				preferJoinType |= preferRightAsHJProbe
 			}
 		default:
 			preferJoinType = p.preferJoinType
 		}
 		p.preferJoinType = preferJoinType
 	}
-	// Clear information from left and right child nodes to prevent multiple calls to this function.
-	p.leftPreferJoinType = 0
-	p.rightPreferJoinType = 0
+	//// Clear information from left and right child nodes to prevent multiple calls to this function.
+	//p.leftPreferJoinType = 0
+	//p.rightPreferJoinType = 0
```

Comment:
Is it intentional to keep them not deleted, or just forget to?

Reply:
I forgot to delete them.